cirq-core/cirq/devices/grid_device_metadata.py

```diff
@@ -79,7 +79,10 @@ def __init__(
             working_gatefamilies = frozenset(gate_durations.keys())
             if working_gatefamilies != gateset.gates:
                 raise ValueError(
-                    "Supplied gate_durations contains gates not present" f" in gateset."
+                    "Supplied gate_durations contains gates not present"
+                    " in gateset."
```

Comment:
nit: This is still not correct, because if `gate_durations` is a subset of gateset, it won't contain any gates not present in the gateset, but vice versa would be true.

Reply:
Updated wording here to reflect the fact we are doing strict equality checks.

Reply:
`gate_durations.keys` should be allowed to be a subset of the gateset, because a gateset could contain GateFamilies that don't represent actual gates, such as `FSimGateFamily`.

An alternative `Gateset` design is to separate out GateFamilies like `FSimGateFamily` into a separate property `equivalence_gate_families`, which are used somewhat differently compared to GateFamilies representing gates.